framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -0,0 +1,1262 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MeshDiagnosticsGenerator.h"
+#include "CastUniquePointer.h"
+
+#include "libmesh/mesh_tools.h"
+#include "libmesh/mesh_refinement.h"
+#include "libmesh/fe.h"
+#include "libmesh/quadrature_gauss.h"
+#include "libmesh/face_tri3.h"
+#include "libmesh/cell_tet4.h"
+#include "libmesh/face_quad4.h"
+#include "libmesh/cell_hex8.h"
+
+registerMooseObject("MooseApp", MeshDiagnosticsGenerator);
+
+InputParameters
+MeshDiagnosticsGenerator::validParams()
+{
+
+  InputParameters params = MeshGenerator::validParams();
+
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to diagnose");
+  params.addClassDescription("Runs a series of diagnostics on the mesh to detect potential issues "
+                             "such as unsupported features");
+
+  // Options for the output level
+  MooseEnum chk_option("NO_CHECK INFO WARNING ERROR", "NO_CHECK");
+
+  params.addParam<MooseEnum>(
```

Comment:
Another check we can add is to make sure a boundary side is covered by a side set. We used to have an issue with Cubit that meshes are not properly merged that ends up with some boundary side in the inside of the domain, which are typically not covered by any side set.

If some checks are more serious, I suggest to change the default for the parameter. For example, that `examine_element_overlap` could be pretty bad, we might want to make it default to warning.

Reply:
it s situation dependent. overlap is totally fine for nodeelem elements for example.

Reply:
the external boundary totally covered is a good idea. Will have to be in the next iteration of adding diagnostics

Reply:
for cubit internal sides, we could add also a "make_sure_these_sidesets_are_external"